projects/packages/forms/src/blocks/field-telephone/edit.js

```diff
@@ -99,6 +103,28 @@ export default function PhoneFieldEdit( props ) {
 		__experimentalCaptureToolbars: true,
 	} );
 
+	// Keep the inner label block's requiredIndicator in sync when it changes
+	const labelClientId = useSelect(
+		select => {
+			const { getBlock } = select( blockEditorStore );
+			const parentBlock = getBlock( clientId );
+			if ( ! parentBlock ) {
+				return undefined;
+			}
+			const labelBlock = parentBlock.innerBlocks.find( block => block.name === 'jetpack/label' );
+			return labelBlock?.clientId;
+		},
+		[ clientId ]
+	);
+
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	useEffect( () => {
+		if ( labelClientId ) {
+			updateBlockAttributes( labelClientId, { requiredIndicator } );
+		}
+	}, [ labelClientId, requiredIndicator, updateBlockAttributes ] );
+
```

Comment:
Fairly sure we're using the same code above for syncing here as well as in field-telephone and in field-textarea files. If so, we could define this once an import it to all three as needed.